spec/components/chart_component_spec.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe ChartComponent, type: :component, include_url_helpers: true do
+  let(:school) { create(:school) }
+  let(:chart_type) { :baseload }
+  let(:params) { { school: school, chart_type: chart_type, html_class: 'usage-chart' } }
+
+  context "with all params" do
+    let(:html) { render_inline(ChartComponent.new(**params)) }
+
+    it { expect(html).to have_selector("div", id: "chart_wrapper_baseload") }
+    it { expect(html).to have_selector("div", class: "usage-chart") }
+
+    it 'sets up chart config data attribute' do
+      expect(html).to have_selector("div", id: 'chart_baseload') { |d| JSON.parse(d['data-chart-config'])['type'] == 'baseload' }
+    end
+  end
+
+  context "with title, subtitle, header and footer slots" do
+    let(:html) do
+      render_inline ChartComponent.new(**params) do |c|
+        c.with_title    { "I'm a title" }
+        c.with_subtitle { "I'm a subtitle" }
+        c.with_header   { "<strong>I'm a header</strong>".html_safe }
+        c.with_footer   { "<small>I'm a footer</small>".html_safe }
+      end
+    end
+    it { expect(html).to have_selector("h4", text: "I'm a title") }
+    it { expect(html).to have_selector("h4", id: "chart_baseload_title") }
+    it { expect(html).to have_selector("h5", text: "I'm a subtitle") }
+    it { expect(html).to have_selector("strong", text: "I'm a header") }
+    it { expect(html).to have_selector("small", text: "I'm a footer") }
```

Comment:
Do you need to test what happens if you don't pass the optional parameters?

Reply:
I've added a test that the h4 and h5 tags aren't rendered if the title and subtitle aren't set. More difficult to test for a missing header or footer, as there's nothing rendered when they're empty.

Do you think that covers it sufficiently?

Reply:
Hadn't thought of those - was more thinking of what happens if school and chart_config aren't passed through for example? They both have defaults of nil so I assume they are optional?